src/question/tests.py

```diff
@@ -1,11 +1,173 @@
+        self.client.force_login(User.objects.get(username="admin"))
+        for i in range(starting_amount + 1, starting_amount + 1 + added_amount):
+            url = "http://127.0.0.1:8000/api/v1/tags/" + str(i) + "/"
+            data = {
+                "url": "/api/v1/tags/" + str(i) + "/",
+                "name": "Tag" + str(i),
+                "slug": "tag" + str(i)
+            }
+            object_count = Tag.objects.count()
+            response = self.client.put(url, data, format="json")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(Tag.objects.count(), object_count + 1)
+        self.client.logout()
+
+    def test_put_mod(self):
+        self.client.force_login(User.objects.get(username="mod"))
+        for i in range(starting_amount + 1, starting_amount + 1 + added_amount):
+            url = "http://127.0.0.1:8000/api/v1/tags/" + str(i) + "/"
+            data = {
+                "url": "/api/v1/tags/" + str(i) + "/",
+                "name": "Tag" + str(i),
+                "slug": "tag" + str(i)
+            }
+            object_count = Tag.objects.count()
+            response = self.client.put(url, data, format="json")
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(Tag.objects.count(), object_count + 1)
+        self.client.logout()
+
+    def test_put_csicska(self):
+        self.client.force_login(User.objects.get(username="csicska"))
+        for i in range(starting_amount + 1, starting_amount + 1 + added_amount):
+            url = "http://127.0.0.1:8000/api/v1/tags/" + str(i) + "/"
+            data = {
+                "url": "/api/v1/tags/" + str(i) + "/",
+                "name": "Tag" + str(i),
+                "slug": "tag" + str(i)
+            }
+            response = self.client.put(url, data, format="json")
+            self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.client.logout()
+
+    def test_put_existing_name(self):
+        self.client.force_login(User.objects.get(username="mod"))
+        url = "http://127.0.0.1:8000/api/v1/tags/6/"
+        data = {
+                "url": "/api/v1/tags/6/",
+                "name": "Tag3",
+                "slug": "tag6"
+            }
+        response = self.client.put(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.client.logout()
+
+    def test_put_existing_slug(self):
+        self.client.force_login(User.objects.get(username="mod"))
+        url = "http://127.0.0.1:8000/api/v1/tags/6/"
+        data = {
+                "url": "/api/v1/tags/6/",
+                "name": "Tag6",
+                "slug": "tag3"
+            }
+        response = self.client.put(url, data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.client.logout()
+
+    def test_overwrite_tag(self):
```

Comment:
This test fails: 
```
Error
Traceback (most recent call last):
  File "/opt/project/src/question/tests.py", line 128, in test_overwrite_tag
    self.assertEqual(Tag.objects.get(pk=1).name, "Tag1")
  File "/usr/local/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 399, in get
    self.model._meta.object_name
taggit.models.Tag.DoesNotExist: Tag matching query does not exist.
```

This is because the "Tag1" tag is not guaranteed to have an id of 1.